src/coreclr/vm/threadstatics.cpp

```diff
@@ -136,7 +136,7 @@ void ThreadLocalBlock::EnsureModuleIndex(ModuleIndex index)
 
     // If this allocation fails, we will throw. If it succeeds,
     // then we are good to go
-    PTR_TLMTableEntry pNewModuleSlots = (PTR_TLMTableEntry) (void*) new BYTE[sizeof(TLMTableEntry) * aModuleIndices];
+    PTR_TLMTableEntry pNewModuleSlots = (PTR_TLMTableEntry) new TLMTableEntry[aModuleIndices];
```

Comment:
```suggestion
    PTR_TLMTableEntry pNewModuleSlots = new TLMTableEntry[aModuleIndices];
```
This cast should be unnecessary now.